Broken update from 0.31.3 to 0.32.4

### Description

**Observed Behavior**:

When trying to update Karpenter from 0.31.3 (and any version before) to 0.32.4, pods are failing into CrashLoopBackOff with
```
panic: parsing settings, invalid character 'm' looking for beginning of value

goroutine 1 [running]:
```

**Expected Behavior**:
Update going successfully

**Reproduction Steps** (Please include YAML):
We have following block in our helm chart values:
```
    settings:
      aws:
        tags:
          sometag: somevalue
```
In 0.31.3 and earlier it renders into following configmap:
```
  aws.tags: '{"sometag":"somevalue"}'
```

In 0.32.4 it looks like this:
```
  aws.tags: map[sometag:somevalue]
```

Obviously Karpenter fails to parse that configuration.

**Versions**:
- Chart Version: 0.32.4
- Kubernetes Version (`kubectl version`): v1.26.11-eks-8cb36c9

--- a/charts/karpenter/templates/configmap.yaml
+++ b/charts/karpenter/templates/configmap.yaml
@@ -50,7 +50,7 @@ data:
   aws.interruptionQueueName: "{{ . }}"
 {{- end }}
 {{- with .Values.settings.aws.tags }}
-  aws.tags: "{{ . }}"
+  aws.tags: {{ . | toJson | quote }}
 {{- end }}
 {{- with .Values.settings.aws.reservedENIs }}
   aws.reservedENIs: "{{ . }}"

